WooCommerce/Classes/ViewRelated/Blaze/ConfirmPayment/BlazeConfirmPaymentViewModel.swift

```diff
@@ -47,10 +54,29 @@ final class BlazeConfirmPaymentViewModel: ObservableObject {
             }
         } catch {
             DDLogError("⛔️ Error fetching payment info for Blaze campaign creation: \(error)")
-            shouldDisplayErrorAlert = true
+            shouldDisplayPaymentErrorAlert = true
         }
         isFetchingPaymentInfo = false
     }
+
+    @MainActor
+    func confirmPaymentDetails() async {
+        guard let selectedPaymentMethod else {
+            DDLogError("⚠️ No payment method found for campaign creation!")
+            return
+        }
+        shouldDisplayCampaignCreationError = false
+        isCreatingCampaign = true
+        do {
+            let updatedDetails = campaignInfo.copy(paymentMethodID: selectedPaymentMethod.id)
+            try await requestCampaignCreation(details: updatedDetails)
+        } catch {
+            DDLogError("⛔️ Error creating Blaze campaign: \(error)")
+            shouldDisplayCampaignCreationError = true
+        }
+        isCreatingCampaign = false
+        completionHandler()
```

Comment:
Nit: Should we call `completionHandler` only upon success?

Reply:
Yes, this was a mistake 😅 Updated in 3a07b222f9d96bc11e572a9ba8584d40f7b465dd.